production/helm/loki/CHANGELOG.md

```diff
@@ -16,6 +16,7 @@ Entries should include a reference to the pull request that introduced the chang
 - [FEATURE] Added readiness probe for memcached [#15609](https://github.com/grafana/loki/pull/15609)
 - [FEATURE] Added pdb for pattern ingester [#17058](https://github.com/grafana/loki/pull/17058)
 - [BUGFIX] Ensure ui.enabled=true is set in loki ConfigMap when loki.ui.enabled=true is set in values.yaml to actually enable the UI [#17562](https://github.com/grafana/loki/pull/17562)
+- [BUGFIX] Fix broken `readinessProbe` and `livenessProbe` helper template definitons. [#18347](https://github.com/grafana/loki/pull/18347)
```

Comment:
If `with` in Helm templates indeed has `else` clauses like in Go templates as you pointed out, then I guess this should be reworded from "Fix broken" to "Streamline" or something like that. WDYT?